Firestore/Example/Tests/Integration/API/FIRQueryTests.mm

```diff
@@ -1180,4 +1190,102 @@ - (void)testOrderByEquality {
                      matchesResult:@[ @"doc6", @"doc3" ]];
 }
 
+- (void)testResumingAQueryShouldUseExistenceFilterToDetectDeletes {
+  // Prepare the names and contents of the 100 documents to create.
+  NSMutableDictionary<NSString *, NSDictionary<NSString *, id> *> *testData =
+      [[NSMutableDictionary alloc] init];
+  for (int i = 0; i < 100; i++) {
+    [testData setValue:@{@"key" : @42} forKey:[NSString stringWithFormat:@"doc%@", @(i)]];
+  }
+
+  // Create 100 documents in a new collection.
+  FIRCollectionReference *collRef = [self collectionRefWithDocuments:testData];
+
+  // Run a query to populate the local cache with the 100 documents and a resume token.
+  NSArray<FIRDocumentReference *> *createdDocuments;
+  {
+    FIRQuerySnapshot *querySnapshot1 = [self readDocumentSetForRef:collRef
+                                                            source:FIRFirestoreSourceDefault];
```

Comment:
check the size of the querySnapshot1 equals to 100, like web and android does:
      `expect(snapshot1.size, 'snapshot1.size').to.equal(100);`
     `assertWithMessage("querySnapshot1").that(querySnapshot.size()).isEqualTo(100);`